### PR TITLE
Allow callers of load_java_class to pass the name of the main function

### PR DESCRIPTION
**Pull request by NathanJPhillips**
This prevents callers forgetting to reset `config.main` if they want the default behaviour.
No CBMC unit test callers are currently setting `config.main`, but as `config` is global, if a caller does set it then without this fix the next test to run will still have it set and will fail to find their main function as it won't match the previously set name.

### CODE REVIEW

**Review by smowton — APPROVED** [review]

File: unit/testing-utils/load_java_class.cpp [hunk 1]
```diff
@@ -28,15 +28,17 @@
 /// \return The symbol table that is generated by parsing this file.
 symbol_tablet load_java_class(
   const std::string &java_class_name,
-  const std::string &class_path)
+  const std::string &class_path,
+  const std::string &main)
```
Comment (thk123) [inline unit/testing-utils/load_java_class.cpp:34]: Update the documentation with an explanation of the default behaviour (`""` for `main`)

File: unit/testing-utils/load_java_class.h [hunk 2]
```diff
@@ -19,11 +19,13 @@
 
 symbol_tablet load_java_class(
   const std::string &java_class_name,
-  const std::string &class_path);
+  const std::string &class_path,
+  const std::string &main = "");
```
Comment (thk123) [inline unit/testing-utils/load_java_class.h:23]: This is a sensible default because `config.main ==  ""` => use the `main` method?

**Review by thk123 — CHANGES_REQUESTED** [review]
Would you consider updating the places that could use this out of scope? If not, could you update them (e.g. `generic_parameters_test` and `GenericNondet`)

In any case, please update the documentation.

File: unit/testing-utils/load_java_class.h (hunk 2 above)
Comment (NathanJPhillips) [reply unit/testing-utils/load_java_class.h:23]: Yes, this is now indicated in the documentation.

**Comment by NathanJPhillips** [comment]
Places this is used are not in CBMC. I have a PR to appropriate repositories that I will push once this is merged.

**Review by thk123 — APPROVED** [review]

**Pull request merged**